master/buildbot/scripts/upgrade_master.py

```diff
@@ -77,6 +78,19 @@ def upgradeDatabase(config, master_cfg):
               % (stripUrlPassword(master_cfg.db['db_url'])))
         print("Warning: Stopping this process might cause data loss")
 
+    def sighandler(signum, frame):
+        msg = " ".join("""
+        WARNING: ignoring signal %s.
+        This process should not be interrupted to avoid database corruption.
+        If you really need to terminate it, use SIGKILL.
+        """.split())
+        print(msg % signum)
+
+    for signame in ("SIGTERM", "SIGINT", "SIGQUIT", "SIGHUP", "SIGUSR1",
+                    "SIGUSR2", "SIGBREAK", "CTRL_C_EVENT", "CTRL_BREAK_EVENT"):
+        if hasattr(signal, signame):
+            signal.signal(getattr(signal, signame), sighandler)
```

Comment:
I guess you need to catch ValueError to ignore error in windows .

Reply:
I think it would be better to know what signal are supported.

I don't have any Windows machine to test this :(

Reply:
Ok, I found a laptop somewhere in a meeting room. Installed python on it and tested :)

`CTRL_C_EVENT` and `CTRL_BREAK_EVENT` are only usable as `os.kill` arguments. They are not to be used to register a signal handler.